### ISSUE TITLE
Spree uses deprecated factory_girl

### ISSUE THREAD

**Issue by mkllnk** [openfoodfoundation/openfoodnetwork#2438]
Follow up on #2262, partly solved by #2266.

## Description

We upgraded from factory_girl to factory_bot, but Spree hasn't. It uses factory_girl even in the current master code. This only affects us, because we use https://github.com/openfoodfoundation/spree/blob/step-6a/core/lib/spree/testing_support/factories.rb in: https://github.com/openfoodfoundation/openfoodnetwork/blob/09534b41e9a806daf91c5633d0b03e009e297b30/spec/factories.rb#L2

## Expected Behavior

No deprecation warnings when running `rspec`.

## Actual Behavior

Output:

> The factory_girl gem has been deprecated and has been replaced by factory_bot. Please switch to factory_bot as soon as possible.
DEPRECATION WARNING: `#ignore` is deprecated and will be removed in 5.0. Please use `#transient` instead. (called from block (2 levels) in <top (required)> at /home/maikel/code/openfoodnetwork/spec/factories.rb:299)

## Steps to Reproduce

1. Run `bundle exec rspec`

## Context

I'm running specs during development and would like to see only important messages.

## Possible Fix

Import Spree's factory code into OFN and change factory_girl to factory_bot. We should check which code we actually use and import only that part. This is another step to become more independent from Spree.

**Comment by Vadlusk** [comment]
Hi there, I'm interested in working on this issue because I'm very enthusiastic about factory-bot, but I'd like to ask a question or two first. I'm unfamiliar with spree, but I can certainly take all of your factories and rewrite them with updated syntax and put them in the normal place(/spec/factories).  It's not the fix you suggested above, but it would make you less dependent on spree and get rid of the deprecation warnings.

**Comment by mkllnk** [comment]
Thank you for your offer @Vadlusk. It's great to see new contributors. :-)

People are working on another task to upgrade Spree at the moment. It will take a while and may be in conflict with this one. We are just discussing if we can change this task or should block it until the Spree upgrade is done.

We would still be happy if you would like to work on this and show us a proof of concept. No promise that we will merge it within the next month though. ;-)

**Comment by Vadlusk** [comment]
No worries, I'm not here to create conflicts, in any fashion. I can choose a different issue within this code base. Thank you for your kind response.

**Comment by sauloperez** [comment]
thanks @Vadlusk ! Feel free to pick up any of the [good first issue](https://github.com/openfoodfoundation/openfoodnetwork/labels/good%20first%20issue) issues

**Comment by luisramos0** [comment]
Removing the reference to #2109 and moving it to epic #2584 (all the issues blocked by the spree upgrade).

**Comment by luisramos0** [comment]
This was recently closed by #5941
